orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -251,24 +255,38 @@ def _get_exported_model(self, *inputs, **kwargs):
         # For example, the Dropout node in a model is dropped under eval mode.
         assert self._export_mode is not None, "Please use a concrete instance of ExecutionManager"
 
+        # Todo: get the flag once wechi provided in later changes.
+        enable_custom_auto_grad = self._onnx_export_type == torch.onnx.OperatorExportTypes.ONNX_FALLTHROUGH
+        exported_model = None
         try:
-            with torch.no_grad(), _logger.suppress_os_stream_output(log_level=self._loglevel):
-                torch.onnx.export(self._flattened_module,
-                                  sample_inputs_as_tuple,
-                                  f,
-                                  input_names=self._input_info.names,
-                                  output_names=output_names,
-                                  opset_version=ONNX_OPSET_VERSION,
-                                  do_constant_folding=False,
-                                  training=self._export_mode,
-                                  dynamic_axes=self._input_info.dynamic_axes,
-                                  verbose=self._loglevel < _logger.LogLevel.WARNING,
-                                  export_params=False,
-                                  keep_initializers_as_inputs=True)
+            if enable_custom_auto_grad:
+                flattened_module_for_export = copy.deepcopy(self._flattened_module)
+            else:
+                flattened_module_for_export = self._flattened_module
+
+            with torch.set_grad_enabled(enable_custom_auto_grad), \
+                _logger.suppress_os_stream_output(log_level=self._loglevel):
+                torch.onnx.export(flattened_module_for_export,
+                                sample_inputs_as_tuple,
+                                f,
+                                input_names=self._input_info.names,
+                                output_names=output_names,
+                                opset_version=ONNX_OPSET_VERSION,
+                                do_constant_folding=False,
+                                training=self._export_mode,
+                                dynamic_axes=self._input_info.dynamic_axes,
+                                verbose=self._loglevel < _logger.LogLevel.WARNING,
+                                export_params=False,
+                                keep_initializers_as_inputs=True,
+                                operator_export_type=self._onnx_export_type)
+                exported_model = onnx.load_model_from_string(f.getvalue())
```

Comment:
I think we can put this line outside of the ```try...except``` and remove the line that initializes ```exported_model = None``` above.

Reply:
Sure.